src/DataStructures/Tensor/Expressions/KroneckerDeltaAsExpression.hpp

```diff
@@ -0,0 +1,174 @@
+  static constexpr size_t num_ops_to_evaluate_primary_right_child = 0;
+  /// If on the primary path, this is the remaining number of arithmetic tensor
+  /// operations that need to be done for this expression's subtree, given that
+  /// we will have already computed the subtree at the next lowest leg's
+  /// starting point. This is just 0 because this expression is a leaf.
+  static constexpr size_t num_ops_to_evaluate_primary_subtree = 0;
+  /// If on the primary path, whether or not the expression is a starting point
+  /// of a leg
+  static constexpr bool is_primary_start = false;
+  /// If on the primary path, whether or not the expression's child along the
+  /// primary path is a subtree that contains a starting point of a leg along
+  /// the primary path. This is always falls because this expression is a leaf.
+  static constexpr bool primary_child_subtree_contains_primary_start = false;
+  /// If on the primary path, whether or not this subtree contains a starting
+  /// point of a leg along the primary path
+  static constexpr bool primary_subtree_contains_primary_start =
+      is_primary_start;
+
+  /// \brief Construct a `KroneckerDeltaAsExpression` from a `KroneckerDelta`
+  ///
+  /// \param k the `KroneckerDelta` to represent as a `TensorExpression`
+  KroneckerDeltaAsExpression(const K& k) : k_(&k) {}
+  ~KroneckerDeltaAsExpression() override = default;
+
+  // This expression does not represent a `Tensor`, nor does it have any
+  // children, so we should never need to assert that the LHS `Tensor` is not
+  // equal to the `KroneckerDelta` stored by this expression
+  template <typename LhsTensor>
+  void assert_lhs_tensor_not_in_rhs_expression(
+      const gsl::not_null<LhsTensor*>) const = delete;
+  // This expression does not represent a `Tensor`, nor does it have any
+  // children, so we should never need to assert that the LHS `Tensor` is not
+  // equal to the `KroneckerDelta` stored by this expression
+  template <typename LhsTensorIndices, typename LhsTensor>
+  void assert_lhs_tensorindices_same_in_rhs(
+      const gsl::not_null<LhsTensor*> lhs_tensor) const = delete;
+
+  /// \brief Returns the value of the contained Kronecker delta's multi-index
+  ///
+  /// \param multi_index the multi-index of the component to retrieve
+  /// \return the value of the component at `multi_index` in the Kronecker delta
+  constexpr SPECTRE_ALWAYS_INLINE double get(
+      const std::array<size_t, num_tensor_indices>& multi_index) const {
+    if (gsl::at(multi_index, 0) != gsl::at(multi_index, 1)) {
+      return 0.0;
+    } else {
+      return 1.0;
+    }
+  }
+
+  /// \brief Returns the value of the contained Kronecker delta's multi-index
+  ///
+  /// \param multi_index the multi-index of the component to retrieve
+  /// \return the value of the component at `multi_index` in the Kronecker delta
+  template <typename ResultType>
+  constexpr SPECTRE_ALWAYS_INLINE double get_primary(
```

Comment:
A general concern: with the `get`s returning things, wouldn't we ultimately want to elide delta's at the `Contract` level? This current means we are doing a _lot_ of math and then multiply by 0. I expected the delta to be used in the other expressions to elide following paths. Without elision this is way too expensive to have.

Reply:
Yah my thoughts were to elide paths where we can in the future implementation of +, -, and * binary operations with other expressions. I implemented the `get` stuff here for completeness.